### ISSUE TITLE
does tapnet need testing data label in inference?

### ISSUE THREAD

**Issue by wz16** [xuczhang/tapnet#14]
Hello Xuchao,
I'm curious on your implementation of TapNet, looks like no matter the model is in training or evaluation mode, it require known data labels to generate the class protocal? I'm assuming in at least labels should be implicit to models at least in testing? Please correct me if I'm wrong...
this is the code line
 https://github.com/xuczhang/tapnet/blob/e4619451b9cf9d3c74aa8320cb591a9da6f02f7d/models.py#L162

Thanks,
Wang

**Comment by fengxy369** [comment]
> Hello Xuchao, I'm curious on your implementation of TapNet, looks like no matter the model is in training or evaluation mode, it require known data labels to generate the class protocal? I'm assuming in at least labels should be implicit to models at least in testing? Please correct me if I'm wrong... this is the code line
> 
> https://github.com/xuczhang/tapnet/blob/e4619451b9cf9d3c74aa8320cb591a9da6f02f7d/models.py#L162
> 
> Thanks, Wang

Do you have an answer to this question now? I'm curious, too

**Comment by wz16** [comment]
> 



> > Hello Xuchao, I'm curious on your implementation of TapNet, looks like no matter the model is in training or evaluation mode, it require known data labels to generate the class protocal? I'm assuming in at least labels should be implicit to models at least in testing? Please correct me if I'm wrong... this is the code line
> > https://github.com/xuczhang/tapnet/blob/e4619451b9cf9d3c74aa8320cb591a9da6f02f7d/models.py#L162
> > 
> > Thanks, Wang
> 
> Do you have an answer to this question now? I'm curious, too


no. i think someone asked the same question in 2020 but not addressed yet.

https://github.com/xuczhang/tapnet/issues/7#issue-679616382
